### PR TITLE
[fix] overlay threshold for sort_by_side

### PR DESCRIPTION
**Pull request by barendgehrels**
Fixes: issue #1186

### CODE REVIEW

File: include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp [hunk 1]
```diff
@@ -325,7 +325,7 @@ public :
                 double
             >::type;
 
-        ct_type const tolerance = 1000000000;
+        static auto const tolerance = common_approximately_equals_epsilon<ct_type>::value();
```
Comment (vissarion) [inline include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp:328]: this is the actual change right?

**Review by vissarion — APPROVED** [review]
Thanks! 

It would help a lot if you comment on the geometric intuition of the value 100 that is used as threshold.

Regarding https://github.com/boostorg/geometry/issues/1183 is there some other combination of values that fixes that or you believe it is of different nature?

File: include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp (hunk 1 above)
Comment (barendgehrels) [reply include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp:328]: Indeed!

Comment (barendgehrels) [reply include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp:328]: I'll document the `100` later. It's not new, it's moved. 
It is recently (earlier this year) adapted from a larger value to this value, considering another issue and all current tests.
It is used for clustering turns, and now also for sort by side. In both cases the exact value is not that important, but it should not be 1 (for floating point comparisons) and not be too large (obviously). The tolerance which was now replaced was very large, I don't remember exactly why.

Comment (barendgehrels) [reply include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp:328]: See also #1108

**Pull request merged**